tools/pythonpkg/setup.py

```diff
@@ -164,9 +165,13 @@ def open_utf8(fpath, flags):
 # make sure we are in the right directory
 os.chdir(os.path.dirname(os.path.realpath(__file__)))
 
-if os.name == 'nt':
-    # windows:
+python_platform = sysconfig.get_platform()
+if os.name == 'nt' and not python_platform.startswith("mingw"):
+    # windows with msvc:
     toolchain_args = ['/wd4244', '/wd4267', '/wd4200', '/wd26451', '/wd26495', '/D_CRT_SECURE_NO_WARNINGS', '/utf-8']
+elif os.name == 'posix' and sys.platform == 'cygwin':
+    # windows with msys or cygwin
+    toolchain_args = ['-std=c++11', '-std=gnu++11', '-g0']
```

Comment:
This doesn't look right - both standard libraries?

Reply:
Yes. For windows. There're 2 different toolchains build for python. One is MSVC (which is used by official build and anaconda build)

Another is GCC (or GNU toolchains), which is used by Cygwin, MingW, and MSYS2.

Reply:
And here's the MSYS2 doc for the environment description: https://www.msys2.org/docs/environments/

https://www.msys2.org/wiki/Porting/